Microsoft.SystemForCrossDomainIdentityManagement/Protocol/ProtocolExtensions.cs

```diff
@@ -102,90 +103,192 @@ private static void Apply(this Core2Group group, PatchOperation2 operation)
                 return;
             }
 
-            OperationValue value;
             switch (operation.Path.AttributePath)
             {
                 case AttributeNames.DisplayName:
-                    value = operation.Value.SingleOrDefault();
-
-                    if (OperationName.Remove == operation.Name)
-                    {
-                        if (
-                            null == value
-                            || string.Equals(group.DisplayName, value.Value, StringComparison.OrdinalIgnoreCase))
-                        {
-                            value = null;
-                        }
-                        else
-                        {
-                            break;
-                        }
-                    }
-
-                    if (null == value)
-                    {
-                        group.DisplayName = null;
-                    }
-                    else
-                    {
-                        group.DisplayName = value.Value;
-                    }
+                    group.DisplayNamePatchOperation(operation);
                     break;
 
                 case AttributeNames.Members:
-                    if (operation.Value != null)
+                    group.MemberPatchOperation(operation);
+                    break;
+            }
+        }
+
+        private static void DisplayNamePatchOperation(this Core2Group group, PatchOperation2 operation)
+        {
+            var value = operation.Value.SingleOrDefault();
```

Comment:
Use strongly typed named wherever possible instead of vars.

string value = operation?.Value?.SingleOrDefault();